meterfs/meterfs.c

```diff
@@ -178,17 +178,21 @@ int meterfs_getFileInfoName(const char *name, fileheader_t *f, meterfs_ctx_t *ct
 {
 	fileheader_t t;
 	uint32_t i, filecnt;
+	ssize_t err = 0;
 
 	meterfs_powerctrl(1, ctx);
 
-	ctx->read(ctx->offset + ctx->hcurrAddr + offsetof(header_t, filecnt), &filecnt, sizeof(filecnt));
+	if ((err = ctx->read(ctx->offset + ctx->hcurrAddr + offsetof(header_t, filecnt), &filecnt, sizeof(filecnt))) < 0)
+		return err;
 
 	for (i = 0; i < filecnt; ++i) {
-		ctx->read(ctx->offset + ctx->hcurrAddr + HGRAIN + (i * HGRAIN) + offsetof(fileheader_t, name), &t.name, sizeof(t.name));
+		if ((err = ctx->read(ctx->offset + ctx->hcurrAddr + HGRAIN + (i * HGRAIN) + offsetof(fileheader_t, name), &t.name, sizeof(t.name))) < 0)
+			return err;
 		if (strncmp(name, t.name, sizeof(t.name)) == 0) {
-			if (f != NULL)
-				ctx->read(ctx->offset + ctx->hcurrAddr + HGRAIN + (i * HGRAIN), f, sizeof(*f));
-
+			if (f != NULL) {
+				if ((err = ctx->read(ctx->offset + ctx->hcurrAddr + HGRAIN + (i * HGRAIN), f, sizeof(*f))) < 0)
+					return err;
```

Comment:
In each of these returns we leave power on. We should always do `meterfs_powerctrl(0, ctx);` before returning.